osquery_rules/linux/osquery_linux_logins_non_office.py

```diff
@@ -15,11 +15,19 @@ def rule(event):
     host_ip = event['columns'].get('host')
     if not host_ip:
         return False
+    host_ipaddr = ipaddress.IPv4Address(host_ip)
 
-    if ipaddress.IPv4Address(host_ip) not in OFFICE_NETWORK.hosts():
-        return True
+    non_office_logins = []
+    for office_network in OFFICE_NETWORKS:
+        # Check the case where the OFFICE_NETWORK is a /32
+        if office_network.num_addresses == 1:
+            non_office_logins.append(
+                office_network.network_address != host_ipaddr)
+        # Check all possible hosts in the office subnet
+        else:
+            non_office_logins.append(host_ipaddr not in office_network.hosts())
 
-    return False
+    return any(non_office_logins)
```

Comment:
If I'm understanding this correctly, we're creating a list of bools and then checking for any True values in that list? Makes more sense (and more efficient) to just return True immediately when we find one:

```python
for office_network in OFFICE_NETWORKS:
    # Check the case where the OFFICE_NETWORK is a /32
    if office_network.num_addresses == 1 and office_network.network_address != host_ipaddr:
        return True
    elif host_ipaddr not in office_network.hosts():
        return True

return False
```

Reply:
I think we actually want to have `return not any(non_office_logins)`, because as long as the IP address is from one of the office IP spaces we're happy right? So maybe:

```
for office_network in OFFICE_NETWORKS:
    # Check the case where the OFFICE_NETWORK is a /32
    if office_network.num_addresses == 1 and office_network.network_address == host_ipaddr:
        return False
    elif host_ipaddr in office_network.hosts():
        return False

return True
```